File: dsa/Patterns/square_hollow.cpp
#include <iostream>
using namespace std;

void square_hollow(int n){
    for (int i = 0; i < n; i++){
        string pattern = "";
        for (int j = 0; j < n; j++){
            if(i == 0 || j == 0 || i == n -1 || j == n - 1){
                pattern += "* ";
            }else{
                pattern += "  ";
            }
        }
        cout << pattern << endl;
    }
}
int main(){
    int n = 5;
    square_hollow(n);
    return 0;
}

File: dsa/Patterns/square_hollow.py
n = 5

for i in range(n):
    pattern = ""
    for j in range(n):
        if i == 0 or j == 0 or i == n-1 or j == n - 1:
            pattern += "* "
        else:
            pattern += " " + " "
    print(pattern)